elastic/beats
# Deprecate aws_partition and use endpoint,regions instead

## What does this PR do?

This PR is to deprecate `aws_partition` config parameter and use `endpoint` with `regions` instead in order to make the AWS configuration less confusing. 

## Why is it important?

There is overlap between `aws_partition` and `endpoint`. We decide to use `endpoint` moving forward.

## Checklist




## How to test this PR locally

1. Test case with no `endpoint` specified
Enable `aws` module:
```
./metricbeat modules enable aws
```

Change `aws.yml` config file to run `ec2` metricset:
```
- module: aws
  period: 5m
  credential_profile_name: elastic-beats
  metricsets:
    - ec2
```
This should work like it used to.

2. Test case with `endpoint`
```
- module: aws
  period: 5m
  credential_profile_name: elastic-beats
  endpoint: amazonaws.com
  regions: us-east-1
  metricsets:
    - ec2
```
Start Metricbeat and `ec2` metricset should work and collect metrics for instances in `us-east-1`.

## Related issues

Closes https://github.com/elastic/beats/issues/23153
